src/main/java/com/accesspoint/rulesengine/entity/Ruleset.java

```diff
@@ -17,7 +14,7 @@
 @Table(name = "ruleset")
 public class Ruleset {
 
-    @Getter @Setter private @Id Long id;
+    @Getter @Setter private @Id @GeneratedValue(strategy=GenerationType.IDENTITY) Long id ;
```

Comment:
What is the data type for this column in the database?

Reply:
In the table, the column was created as BIGSERIAL. Originally, we did not have the @GeneratedValue annotation, and using SQL insert worked great for adding a new entry. The issue was when we tried to save a new object to our rulesetRepository and having it add it that way, it would throw errors which adding the annotation fixed.